### PR TITLE
Minor refactors in HiveClientModule and MetastoreUtil

### PR DESCRIPTION
**Pull request by shixuan-fan**
This is a follow up PR to address additional comments in #12386 after merging.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadataFactory.java [hunk 1]
```diff
@@ -130,7 +129,7 @@ public HiveMetadataFactory(
         this.locationService = requireNonNull(locationService, "locationService is null");
         this.tableParameterCodec = requireNonNull(tableParameterCodec, "tableParameterCodec is null");
         this.partitionUpdateCodec = requireNonNull(partitionUpdateCodec, "partitionUpdateCodec is null");
-        this.fileRenameExecutor = listeningDecorator(requireNonNull(executorService, "executorService is null"));
+        this.fileRenameExecutor = listeningDecorator(requireNonNull(fileRenameExecutor, "executorService is null"));
```
Comment (arhimondr) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadataFactory.java]: You don't need to wrap it with the `listeningDecorator` again

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveStagingFileCommitter.java [hunk 2]
```diff
@@ -43,7 +42,7 @@
     @Inject
     public HiveStagingFileCommitter(
             HdfsEnvironment hdfsEnvironment,
-            @ForFileRename ExecutorService executorService)
+            @ForFileRename ListeningExecutorService executorService)
     {
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.commitExecutor = listeningDecorator(executorService);
```
Comment (arhimondr) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveStagingFileCommitter.java]: You don't need to wrap it again.

Comment (arhimondr) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveStagingFileCommitter.java]: Add a `requireNonNull`

File: presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java [hunk 3]
```diff
@@ -747,9 +748,9 @@ protected final void setup(String databaseName, HiveClientConfig hiveClientConfi
                 locationService,
                 new TableParameterCodec(),
                 partitionUpdateCodec,
-                newFixedThreadPool(2),
+                listeningDecorator(newFixedThreadPool(2)),
```
Comment (arhimondr) [inline presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java]: Maybe just use the existing `executor`

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java [hunk 4]
```diff
@@ -141,11 +145,11 @@ public ExecutorService createCachingHiveMetastoreExecutor(HiveConnectorId hiveCl
     @ForFileRename
     @Singleton
     @Provides
-    public ExecutorService createFileRanemeExecutor(HiveConnectorId hiveClientId, HiveClientConfig hiveClientConfig)
+    public ListeningExecutorService createFileRanemeExecutor(HiveConnectorId hiveClientId, HiveClientConfig hiveClientConfig)
     {
-        return newFixedThreadPool(
-                hiveClientConfig.getMaxConcurrentFileRenames(),
-                daemonThreadsNamed("hive-file-rename-" + hiveClientId + "-%s"));
+        return listeningDecorator(new ExecutorServiceAdapter(new BoundedExecutor(newCachedThreadPool(
```
Comment (arhimondr) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java]: nit: Maybe format it as

```
listeningDecorator(
  new ExecutorServiceAdapter(
     new BoundedExecutor(
        newCachedThreadPool(
```

**Comment by wenleix** [comment]
I would take a look into this tomorrow but it should be good :)

**Pull request merged**